src/model/ki3700.rs

```diff
@@ -0,0 +1,995 @@
+        })
+    }
+
+    #[must_use]
+    pub const fn new(interface: Box<dyn Interface>, auth: Box<dyn Authentication>) -> Self {
+        Self {
+            info: None,
+            interface,
+            auth,
+        }
+    }
+
+    pub fn add_info(&mut self, info: InstrumentInfo) -> &Self {
+        self.info = Some(info);
+        self
+    }
+}
+
+//Implement device_interface::Interface since it is a subset of instrument::Instrument trait.
+impl instrument::Instrument for Instrument {}
+
+fn is_3700(model: impl AsRef<str>) -> bool {
+    [
+        "3706",
+        "3706-SNFP",
+        "3706-S",
+        "3706-NFP",
+        "3706A",
+        "3706A-SNFP",
+        "3706A-S",
+        "3706A-NFP",
+        "707B",
+        "708B",
+    ]
+    .contains(&model.as_ref())
+}
+
+impl Info for Instrument {
+    fn info(&mut self) -> crate::error::Result<InstrumentInfo> {
+        if let Some(inst_info) = self.info.clone() {
+            return Ok(inst_info);
+        }
+
+        get_info(self)
+    }
+}
+
+impl language::Language for Instrument {}
+
+impl Login for Instrument {
+    fn check_login(&mut self) -> crate::error::Result<instrument::State> {
+        self.write_all(b"print('unlocked')\n")?;
+        for _i in 0..5 {
+            std::thread::sleep(Duration::from_millis(100));
+            let mut resp: Vec<u8> = vec![0; 256];
+            let _read = self.read(&mut resp)?;
+            let resp = std::str::from_utf8(resp.as_slice())
+                .unwrap_or("")
+                .trim_matches(char::from(0))
+                .trim();
+
+            if resp.contains("unlocked") {
+                return Ok(instrument::State::NotNeeded);
+            }
+            if resp.contains("Port in use") {
+                return Ok(instrument::State::LogoutNeeded);
+            }
+        }
+
+        Ok(instrument::State::Needed)
+    }
+
+    fn login(&mut self) -> crate::error::Result<()> {
+        let mut inst_login_state = self.check_login()?;
+        if instrument::State::NotNeeded == inst_login_state {
+            return Ok(());
+        } else if instrument::State::LogoutNeeded == inst_login_state {
+            return Err(InstrumentError::InterfaceLoginErr);
+        }
+
+        println!("Instrument might be locked.\nEnter the password to unlock the instrument:");
+        let password = self.auth.read_password()?;
+        self.write_all(format!("password {password}\n").as_bytes())?;
+
+        inst_login_state = self.check_login()?;
+        if instrument::State::NotNeeded == inst_login_state {
+            println!("Login successful.");
+        } else if instrument::State::Needed == inst_login_state {
+            return Err(InstrumentError::LoginRejected);
+        }
+
+        Ok(())
+    }
+}
+
+impl Script for Instrument {}
+
+impl Flash for Instrument {
+    fn flash_firmware(&mut self, image: &[u8], _: Option<u16>) -> crate::error::Result<()> {
+        let mut image = image.reader();
```

Comment:
If this is the same as TTI, please add a comment stating that within this function.

Reply:
it is not notably:
` localnode.prompts= 0`
is not the same as the tti `localnode.prompts=localnode.DISABLE`

and i removed the command:
`if ki.upgrade ~= nil and ki.upgrade.noacklater ~= nil then ki.upgrade.noacklater() end`

as   `ki.upgrade.noacklater ~= nil` 
returns false so no need for that statement 

otherwise the commands are the same